### PR TITLE
Fluent RowsAffected/LastInsertId builder for insert/update/delete

### PR DESCRIPTION
**Pull request by shaxbee**
Example usage:
```go
count, err := Delete("test").RowsAffected().RunWith(db).ExecContext(ctx)
```

### CODE REVIEW

**Comment by shaxbee** [comment]
@elgris could you kindly review?

**Comment by elgris** [comment]
Hey there @shaxbee !
Sorry for late response, as usual, busy with lots of stuff :)

The change looks rather opinionated. The arguments are:
1. This library generates SQL, but the change moves it to direction of running the sql. Ideally, I'd rather delegate this responsibility to the user of the library.
2. Call chain becomes inconsistent. Example:
```go
Insert("test")     // returns *InsertBuilder
  .Values(1)         // returns *InsertBuilder
  .RunWith(db)   // returns *InsertBuilder
  .LastInsertId()  // BOOM! returns ResultBuilder
  .Exec()              // Terminal operation, returns result
```
3. To me it looks like if you want to wrap the builder into runner, then you should really wrap it, not make a change in the call chain:
```go
wrapWithLastInsertIdExecer(
  Insert("test")     // returns *InsertBuilder
    .Values(1)         // returns *InsertBuilder
    .RunWith(db)   // returns *InsertBuilder
    .LastInsertId()  // still returns *InsertBuilder
).Exec() // returns int64, error
```

What do you think?

**Comment by shaxbee** [comment]
Yes, it is transforming builder but at the same time it eliminates need for double error checking in many cases encountered by users of sqrl.

If this solution is unacceptable for you could we provide at least top-level functions?
```go
c, err := sq.RowsAffected(sq.Insert("test").
    Values(1).
    RunWith(db).
    ExecContext(ctx),
)
```

File: result.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package sqrl
+
+import (
+	"database/sql"
+)
+
+func RowsAffected(res sql.Result, err error) (int64, error) {
```
Comment (elgris) [inline result.go]: Please add a comment, especially that "`err` is an error from query execution". An example is also welcome :)

**Comment by elgris** [comment]
It's much smaller PR now, looks great, thanks :)
Just one comment there, and we can merge it!

**Pull request merged**